autogpt_platform/backend/backend/blocks/codex.py

```diff
@@ -0,0 +1,280 @@
+            advanced=True,
+        )
+        model: CodexModel = SchemaField(
+            title="Codex Model",
+            default=CodexModel.GPT5_1_CODEX,
+            description="Codex-optimized model served via the Responses API.",
+            advanced=False,
+        )
+        reasoning_effort: CodexReasoningEffort = SchemaField(
+            title="Reasoning Effort",
+            default=CodexReasoningEffort.MEDIUM,
+            description="Controls the Responses API reasoning budget. Select 'none' to skip reasoning configs.",
+            advanced=True,
+        )
+        max_output_tokens: int | None = SchemaField(
+            title="Max Output Tokens",
+            default=2048,
+            description="Upper bound for generated tokens (hard limit 128,000). Leave blank to let OpenAI decide.",
+            advanced=True,
+        )
+        credentials: CodexCredentials = CodexCredentialsField()
+
+    class Output(BlockSchemaOutput):
+        response: str = SchemaField(
+            description="Code-focused response returned by the Codex model."
+        )
+        reasoning: str = SchemaField(
+            description="Reasoning summary returned by the model, if available.",
+            default="",
+        )
+        response_id: str = SchemaField(
+            description="ID of the Responses API call for auditing/debugging.",
+            default="",
+        )
+
+    def __init__(self):
+        super().__init__(
+            id="86a2a099-30df-47b4-b7e4-34ae5f83e0d5",
+            description="Generate or refactor code using OpenAI's Codex (Responses API).",
+            categories={BlockCategory.AI, BlockCategory.DEVELOPER_TOOLS},
+            input_schema=CodexBlock.Input,
+            output_schema=CodexBlock.Output,
+            test_input=[
+                {
+                    "prompt": "Write a TypeScript function that deduplicates an array.",
+                    "credentials": TEST_CREDENTIALS_INPUT,
+                }
+            ],
+            test_output=[
+                ("response", str),
+                ("reasoning", str),
+                ("response_id", str),
+            ],
+            test_mock={
+                "call_codex": lambda *args, **kwargs: {
+                    "response": "function dedupe<T>(items: T[]): T[] { return [...new Set(items)]; }",
+                    "reasoning": "Used Set to remove duplicates in O(n).",
+                    "response_id": "resp_test",
+                }
+            },
+            test_credentials=TEST_CREDENTIALS,
+        )
+        self.execution_stats = NodeExecutionStats()
+
+    async def call_codex(
+        self,
+        *,
+        credentials: APIKeyCredentials,
+        model: CodexModel,
+        prompt: str,
+        system_prompt: str,
+        max_output_tokens: int | None,
+        reasoning_effort: CodexReasoningEffort,
+    ) -> dict[str, Any]:
```

Comment:
you can ask AI to type the return value so we have named dict or an actual object or tuple